### DISCUSSION TITLE
Do not know how to serialize a BigInt

### DISCUSSION THREAD

**Discussion by inezabonte** [prisma/prisma#9793] [Q&A] [answered]
Trying to fetch data from the Contacts table but I keep getting the error `Do not know how to serialize a BigInt`.
I'm fetching the data from an server-less function in Next.js. Any ideas on how to solve this?
```prisma
model Contacts {
  ContactID       BigInt   @id @unique(map: "ContactID") @default(autoincrement()) @db.UnsignedBigInt
}
```

**Comment by ryands17** [comment, marked as answer]
@inezabonte 👋 

Have a look at [this doc](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt#use_within_json) to serialise BigInts

**Reply by inezabonte** [reply]
Thanks, this fixed it for me

**Reply by Timoumi-Mahmoud** [reply]
hello, I am using prisma with mysql  and i have some fields that have Bigint type after reading the link that you have provided I resolved the problem with : 
BigInt.prototype.toJSON = function () {
  return this.toString();
};
but if every time I will interact with a database I  will define this function  I will be copying and past those lines , is there is a way to get rid of this repetition  ?
Thank you.

**Reply by jhwheeler** [reply]
@Timoumi-Mahmoud I solved this by using a PrismaClient extension. So you can modify your client initialization code: 

```
  return new PrismaClient().$extends({
    /**
     * Prisma Client returns records as POJOs, and if you attempt to stringify
     * an object w/ a `BigInt` field, the client will return an error;
     * for that reason, we need to cast it to a string.
     * @see https://www.prisma.io/docs/orm/prisma-client/special-fields-and-types#serializing-bigint
     */
    query: {
      $allOperations: async ({ args, query }) => {
        const result = await query(args);
        return JSON.parse(
          JSON.stringify(result, (key, value) =>
            typeof value === 'bigint' ? value.toString() : value,
          ),
        );
      },
    },
  });
```

Here are the docs on the `query` extension: https://www.prisma.io/docs/orm/prisma-client/client-extensions/query

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
